### PR TITLE
fix(deis-cli): use POSIX compatible sh

### LINKED ISSUE #195: make install-v2.sh work with /bin/sh
There are certain bash-isms in `install-v2.sh` that makes it such that it only works when run under `/bin/bash`. We should strive to make it portable with `/bin/sh` if possible.

### PR DESCRIPTION
**Pull request by sgoings**
- fixes #195.

Attempt number 2. Critical changes (vs. #196) @ line 19 + 40

### CODE REVIEW

File: deis-cli/install-v2.sh [hunk 1]
```diff
@@ -1,44 +1,43 @@
...
+# - download deis cli binary
+# - making sure deis cli binary is executable
 # - explain what was done
 #
 
 # install current version unless overridden by first command-line argument
 VERSION=${1:-stable}
 
-set -eo pipefail -o nounset
+set -euf
```
Comment (bacongobbler) [inline deis-cli/install-v2.sh]: how come we need to add the noglob option (`-f`) and unset pipefail? Is pipefail unsupported by /bin/sh?

http://www.tldp.org/LDP/abs/html/options.html

Comment (bacongobbler) [inline deis-cli/install-v2.sh]: e.g. I see

```
$ /bin/sh -c 'set --help'
/bin/sh: line 0: set: --: invalid option
set: usage: set [-abefhkmnptuvxBCHP] [-o option-name] [--] [arg ...]
```

Comment (sgoings) [inline deis-cli/install-v2.sh]: Good catch re: the changes here. 

However, what you executed there with set usage is actually showing usage of bash's `set`. (I'm assuming you're on OSX):

```
$ /bin/sh --help
GNU bash, version 3.2.57(1)-release-(x86_64-apple-darwin16)
Usage:	/bin/sh [GNU long option] [option] ...
	/bin/sh [GNU long option] [option] script-file ...
GNU long options:
	--debug
	--debugger
	--dump-po-strings
	--dump-strings
	--help
	--init-file
	--login
	--noediting
	--noprofile
	--norc
	--posix
	--protected
	--rcfile
	--restricted
	--verbose
	--version
	--wordexp
Shell options:
	-irsD or -c command or -O shopt_option		(invocation only)
	-abefhkmnptuvxBCHP or -o option
Type `/bin/sh -c "help set"' for more information about shell options.
Type `/bin/sh -c help' for more information about shell builtin commands.
Use the `bashbug' command to report bugs.
```

On ubuntu, `/bin/sh` is actually `dash`. There's no pipefail feature supported in dash and there's not even a `-o` option 😞!

Have any suggestions to address your concerns given the information above?

Comment (bacongobbler) [inline deis-cli/install-v2.sh]: Nope! Considering that we don't perform any piping, I don't think there's anything to be concerned about here. I was just curious about the reason for the change. Thanks for clarifying!

**Pull request merged**